src/main/java/com/google/api/codegen/metacode/InitCodeNode.java

```diff
@@ -306,10 +307,12 @@ private static void validateType(
           throw new IllegalArgumentException(
               "typeRef " + typeRef + " not compatible with " + lineType);
         }
-        for (int i = 0; i < childKeys.size(); i++) {
-          if (!childKeys.contains(Integer.toString(i))) {
-            throw new IllegalArgumentException(
-                "typeRef " + typeRef + " must have ordered indices, got " + childKeys);
+        if (typeRef.getApiSource().equals(ApiSource.PROTO)) {
```

Comment:
Why is this block not done for Discovery? I'd really like to avoid source-specific logic in the init code logic.

Reply:
Whatever problem that hack had punted has gone away!
Removed.